docroot/modules/custom/custom_contact/src/Entity/Contact.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\custom_contact\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\custom_contact\ContactInterface;
+
+/**
+ * Defines the contact entity class.
+ *
+ * @ContentEntityType(
+ *   id = "contact",
+ *   label = @Translation("Contact"),
+ *   label_collection = @Translation("Contacts"),
+ *   handlers = {
+ *     "view_builder" = "Drupal\custom_contact\ContactViewBuilder",
+ *     "list_builder" = "Drupal\custom_contact\ContactListBuilder",
+ *     "views_data" = "Drupal\views\EntityViewsData",
+ *     "access" = "Drupal\custom_contact\ContactAccessControlHandler",
+ *     "form" = {
+ *       "add" = "Drupal\custom_contact\Form\ContactForm",
+ *       "edit" = "Drupal\custom_contact\Form\ContactForm",
+ *       "delete" = "Drupal\Core\Entity\ContentEntityDeleteForm"
+ *     },
+ *     "route_provider" = {
+ *       "html" = "Drupal\Core\Entity\Routing\AdminHtmlRouteProvider",
+ *     }
+ *   },
+ *   base_table = "contact",
+ *   admin_permission = "administer contact",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "uuid" = "uuid"
+ *   },
+ *   links = {
+ *     "add-form" = "/admin/content/contact/add",
+ *     "canonical" = "/contact/{contact}",
+ *     "edit-form" = "/admin/content/contact/{contact}/edit",
+ *     "delete-form" = "/admin/content/contact/{contact}/delete",
+ *     "collection" = "/admin/content/contact"
+ *   },
+ *   field_ui_base_route = "entity.contact.settings"
+ * )
+ */
+class Contact extends ContentEntityBase implements ContactInterface {
```

Comment:
Is this entity revisionable?

Reply:
It is fieldable and not revisionable.
Base fields are name, email, telephone and address.